hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/ProtoConversionUtil.java

```diff
@@ -80,17 +83,19 @@ public static GenericRecord convertToAvro(Schema schema, Message message) {
    * 2. Convert directly from a protobuf {@link Message} to a {@link GenericRecord} while properly handling enums and wrapped primitives mentioned above.
    */
   private static class AvroSupport {
+    private static final Schema STRING_SCHEMA = Schema.create(Schema.Type.STRING);
+    private static final Schema NULL_SCHEMA = Schema.create(Schema.Type.NULL);
+    private static final String OVERFLOW_DESCRIPTOR_FIELD_NAME = "descriptor_full_name";
+    private static final String OVERFLOW_BYTES_FIELD_NAME = "proto_bytes";
+    private static final Schema RECURSION_OVERFLOW_SCHEMA = Schema.createRecord("recursion_overflow", null, "org.apache.hudi.proto", false,
```

Comment:
We're not renaming the field, it's just a name for the schema/type. You can see the output [here](https://github.com/apache/hudi/pull/6761/files#diff-c4931766d3118fdebd1fb97fce4b5c36923f6bd66f6226adbfba1dd7ec2ddbd4R56)

Reply:
L59?  https://github.com/apache/hudi/pull/6761/files#diff-c4931766d3118fdebd1fb97fce4b5c36923f6bd66f6226adbfba1dd7ec2ddbd4R59

Reply:
Yes, that's where the name of the type is coming through. Remember that the name of a record and the name of a field are different but both use "name" in the avro schema definition. Line 56 is the field name. The record name is more like a class name in java where it can be defined once and reused for different variables. You can see this down at [line 141](https://github.com/apache/hudi/pull/6761/files#diff-c4931766d3118fdebd1fb97fce4b5c36923f6bd66f6226adbfba1dd7ec2ddbd4R141) where the type is simply `"type" : [ "null", "org.apache.hudi.proto.recursion_overflow" ]`

Reply:
ah, got it.  thank you for the explanation.